ESP-IDF-Subordinate/main/file.c

```diff
@@ -0,0 +1,133 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/queue.h"
+#include "freertos/semphr.h"
+#include "esp_now.h"
+
+#include "file.h"
+
+
+static struct{
+
+    uint8_t dev_mac_address[6];
+    bool disable_sleep;
+    uint32_t tx_nack_timeout_ms;
+    data_to_transmit _data_to_transmit;
+    on_data_rx _on_data_rx;
+    sub_transmit_periods_t *transmit_periods;
+    QueueHandle_t rx_data;
+    SemaphoreHandle_t sleep_lock;
+    uint8_t *tx_buffer;
+    uint8_t tx_data_size;
+    esp_now_send_status_t last_message_status;
+    uint16_t interval_ms;
+    SemaphoreHandle_t mutex;
+
+} sub_struct;
+
+sub_status_t init(sub_init_struct_t *init_struct, sub_transmit_periods_t *transmit_periods){
+    
+    if(init_struct == NULL || transmit_periods == NULL){
+        return SUB_NULL_ERR;
+    }
+
+    sub_struct.disable_sleep = init_struct->disable_sleep;
+
+    if(init_struct->tx_nack_timeout_ms > 60000){
+
+        return SUB_TIMES_ERR;
+    }
+
+    sub_struct.tx_nack_timeout_ms = init_struct->tx_nack_timeout_ms;
+
+    if(init_struct->_data_to_transmit == NULL){
+
+        return SUB_NULL_ERR;
+    }
+
+    sub_struct._data_to_transmit = init_struct->_data_to_transmit;
+
+
+    if (transmit_periods->init_ms == 0 || transmit_periods->idle_ms == 0 || transmit_periods->armed_ms == 0 || 
```

Comment:
Widzę że zdedcydowaliśmy się na strukture, a chyba jednak lepiej by było zmienić to na tabilcę tych czasów, a w enumie stany. To sprawdzanie trzeba będzie zmienić na pętle, ale później w przypadku zmian stanów, zmieniamy tylko enum i tablicę, a tutaj będzie się sprawdzać autoamtycznie w pętli.
```
enum {
INIT_MS = 0,
IDLE_MS
...
ENUM_MAX
};

transmit_periods[ENUM_MAX]
```